sympy/assumptions/refine.py

```diff
@@ -336,6 +336,29 @@ def refine_sign(expr, assumptions):
     return expr
 
 
+def refine_matrixelement(expr, assumptions):
+    """
+    Handler for symmetric part
+
+    Examples
+    ========
+
+    >>> from sympy.assumptions.refine import refine_matrixelement
+    >>> from sympy import Q
+    >>> from sympy.matrices.expressions.matexpr import MatrixSymbol
+    >>> X = MatrixSymbol('X', 3, 3)
+    >>> refine_matrixelement(X[0, 1], Q.symmetric(X))
+    X[0, 1]
+    >>> refine_matrixelement(X[1, 0], Q.symmetric(X))
+    X[0, 1]
+    """
+    from sympy.matrices.expressions.matexpr import MatrixElement
+    matrix, i, j = expr.args
+    if ask(Q.symmetric(matrix), assumptions):
+        if ask(Q.positive(i - j)):
```

Comment:
Maybe it can be better to use sort and permute them lexicographical order, than computing if i is bigger than j.
It should have better support symbolic dimensions

Reply:
On the issue I suggested to use `(i - j).could_extract_minus_sign()`. I don't know if it would be surprising if refine ignored assumptions on i and j. I guess there is no guarantee that it gives the smaller element first. It just needs to be canonical. The key thing is that something like `X[i, j] - X[j, i]` should cancel.